fenix/app/src/main/java/org/mozilla/fenix/tabstray/browser/compose/ReorderableList.kt

```diff
@@ -0,0 +1,270 @@
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.graphicsLayer
+import androidx.compose.ui.hapticfeedback.HapticFeedback
+import androidx.compose.ui.hapticfeedback.HapticFeedbackType
+import androidx.compose.ui.input.pointer.pointerInput
+import androidx.compose.ui.platform.LocalHapticFeedback
+import androidx.compose.ui.platform.LocalViewConfiguration
+import androidx.compose.ui.zIndex
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.launch
+
+/**
+ * Remember the reordering state for reordering list items
+ *
+ * @param listState State of the list
+ * @param onMove Callback to be invoked when switching between two items
+ * @param ignoredItems List of keys for non-draggable items
+ * @param onLongPress Optional callback to be invoked when long pressing an item
+ * @param onExitLongPress Optional callback to be invoked when the item is dragged after long press
+ */
+@Composable
+fun createListReorderState(
+    listState: LazyListState,
+    onMove: (LazyListItemInfo, LazyListItemInfo) -> Unit,
+    ignoredItems: List<Any>,
+    onLongPress: (LazyListItemInfo) -> Unit = {},
+    onExitLongPress: () -> Unit = {},
+): ListReorderState {
+    val scope = rememberCoroutineScope()
+    val touchSlop = LocalViewConfiguration.current.touchSlop
+    val hapticFeedback = LocalHapticFeedback.current
+    val state = remember(listState) {
+        ListReorderState(
+            listState = listState,
+            onMove = onMove,
+            scope = scope,
+            touchSlop = touchSlop,
+            hapticFeedback = hapticFeedback,
+            ignoredItems = ignoredItems,
+            onLongPress = onLongPress,
+            onExitLongPress = onExitLongPress,
+        )
+    }
+    return state
+}
+
+/**
+ * Class containing details about the current state of dragging in list
+ *
+ * @param listState State of the list
+ * @param scope [CoroutineScope] used for scrolling to the target item
+ * @param hapticFeedback [HapticFeedback] used for performing haptic feedback on item long press
+ * @param touchSlop Distance in pixels the user can wander until we consider they started dragging
+ * @param onMove Callback to be invoked when switching between two items
+ * @param onLongPress Optional callback to be invoked when long pressing an item
+ * @param onExitLongPress Optional callback to be invoked when the item is dragged after long press
+ * @param ignoredItems List of keys for non-draggable items
+ */
+class ListReorderState internal constructor(
+    private val listState: LazyListState,
+    private val scope: CoroutineScope,
+    private val hapticFeedback: HapticFeedback,
+    private val touchSlop: Float,
+    private val onMove: (LazyListItemInfo, LazyListItemInfo) -> Unit,
+    private val onLongPress: (LazyListItemInfo) -> Unit = {},
+    private val onExitLongPress: () -> Unit = {},
+    private val ignoredItems: List<Any> = emptyList(),
```

Comment:
why are these properties optional?

Reply:
My initial thought was these properties are more specific to Fenix tabs and thus could be optional. I can change them to be mandatory.